unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -1,43 +1,210 @@
             dataStore.exploreV2.configureBackpackInFullscreenMenu.OnChange += ConfigureBackpackInFullscreenMenuChanged;
             ConfigureBackpackInFullscreenMenuChanged(dataStore.exploreV2.configureBackpackInFullscreenMenu.Get(), null);
+            dataStore.exploreV2.isOpen.OnChange += ExploreV2IsOpenChanged;
+            backpackEmotesSectionController.OnNewEmoteAdded += OnNewEmoteAdded;
+            backpackEmotesSectionController.OnEmotePreviewed += OnEmotePreviewed;
+            backpackEmotesSectionController.OnEmoteEquipped += OnEmoteEquipped;
+            backpackEmotesSectionController.OnEmoteUnequipped += OnEmoteUnequipped;
             SetVisibility(dataStore.HUDs.avatarEditorVisible.Get());
         }
 
         public void Dispose()
         {
+            ownUserProfile.OnUpdate -= LoadUserProfile;
             dataStore.HUDs.avatarEditorVisible.OnChange -= SetVisibility;
             dataStore.exploreV2.configureBackpackInFullscreenMenu.OnChange -= ConfigureBackpackInFullscreenMenuChanged;
+            dataStore.exploreV2.isOpen.OnChange -= ExploreV2IsOpenChanged;
+
+            backpackEmotesSectionController.OnNewEmoteAdded -= OnNewEmoteAdded;
+            backpackEmotesSectionController.OnEmotePreviewed -= OnEmotePreviewed;
+            backpackEmotesSectionController.OnEmoteEquipped -= OnEmoteEquipped;
+            backpackEmotesSectionController.OnEmoteUnequipped -= OnEmoteUnequipped;
+            backpackEmotesSectionController.Dispose();
+
             view.Dispose();
         }
 
         public void SetVisibility(bool visible)
         {
             if (visible)
+            {
+                backpackEmotesSectionController.LoadEmotes();
+                LoadUserProfile(ownUserProfile, true);
                 view.Show();
+            }
             else
+            {
                 view.Hide();
+                view.ResetPreviewEmote();
+            }
+
+            SetRenderScale(visible);
         }
 
         private void SetVisibility(bool current, bool _) =>
             SetVisibility(current);
 
         private void ConfigureBackpackInFullscreenMenuChanged(Transform currentParentTransform, Transform previousParentTransform) =>
             view.SetAsFullScreenMenuMode(currentParentTransform);
+
+        private void ExploreV2IsOpenChanged(bool current, bool previous)
+        {
+            if (current || !avatarIsDirty)
+                return;
+
+            avatarIsDirty = false;
+            LoadUserProfile(ownUserProfile, true);
+            backpackEmotesSectionController.RestoreEmoteSlots();
+        }
+
+        private void LoadUserProfile(UserProfile userProfile) =>
+            LoadUserProfile(userProfile, false);
+
+        private void LoadUserProfile(UserProfile userProfile, bool forceLoading)
+        {
+            bool avatarEditorNotVisible = CommonScriptableObjects.rendererState.Get() && !view.isVisible;
+            bool isPlaying = !Application.isBatchMode;
```

Comment:
whats the case of usage for this line? we should try to avoid this reference

Reply:
Removed for now: https://github.com/decentraland/unity-renderer/pull/4933/commits/739ce52464ebbfcc85b624e8b5f1e696e602b982